File: app/src/main/java/com/sewon/officehealth/service/ble/SerialListener.java
package com.sewon.officehealth.service.ble;

import java.util.ArrayDeque;

public interface SerialListener {
  void onSerialConnect();

  void onSerialConnectError(Exception e);

  void onSerialRead(byte[] data);                // socket -> service

  void onSerialRead(ArrayDeque<byte[]> datas);   // service -> UI thread

  void onSerialIoError(Exception e);
}

File: app-office/src/main/java/com/sewon/officehealth/service/ble/SerialListener.kt
package com.sewon.officehealth.service.ble

import java.util.ArrayDeque

interface SerialListener {
  fun onSerialConnect()
  fun onSerialConnectError(e: Exception)
  fun onSerialRead(data: ByteArray) // socket -> service
  fun onSerialRead(datas: ArrayDeque<ByteArray>) // service -> UI thread
  fun onSerialIoError(e: Exception)
}